lib/jit.ml

```diff
@@ -213,19 +213,18 @@ let backend = (module Backend : Backend_intf.S)
 
 let jit_load_body ppf phrase_name program =
   let open Config in
-  let dll =
-    if !Clflags.keep_asm_file then phrase_name ^ ext_dll
-    else Filename.temp_file ("caml" ^ phrase_name) ext_dll
+  let prefixname =
+    if !Clflags.keep_asm_file then phrase_name
+    else Filename.concat Filename.temp_dir_name ("caml" ^ phrase_name)
```

Comment:
`Filename.temp_dir_name` is deprecated according to the OCaml documentation.

This code has different semantics, because `temp_dir_name` unlike `temp_file` does not create an uniquely named, non-clashing filename, instead it just returns the temporary folder, so most of the time it would end up being something like `/tmp/caml_$phrase_name`. Oddly enough `Filename` doesn't seem to provide an option for creating a temporary directory.

Reply:
Judging from the output of `strace`, this path is not used for actual disk io, so perhaps just `phrase_name` would be even simpler? Alternatively I could change it to `Filename.get_temp_dir_name`. Or, to be more robust, inline the implementation of the tempfile name generation:

```ocaml
  let rnd = (Random.bits ()) land 0xFFFFFF in
  concat temp_dir (Printf.sprintf "%s%06x%s" prefix rnd suffix)
```